### PR TITLE
NXAGENT_UPGRADE code cleanup:

### PR DESCRIPTION
**Pull request by sunweaver**
NX agent contains/ed two build trees. An old one (probably pre-3.x.y)
 and a "newer" one. The "newer" code tree used to become enabled by
 setting NXUpgradeAgentServer in nx-X11/config/cf/host.def to YES.

 As building the NXUpgradeAgentServer has been the default for
 years now, we drop all code that does not get used at build time
 for NXUpgradeAgentServer == YES (i.e., the code that belongs to the
 pre-3.x.y phase of NX agent).

NOTES FOR THE REVIEWER...

This commit contains the following steps:
   o remove all references of NXUpgradeAgentServer and -DNXAGENT_UPGRADE
   o remove code blocks encompassed by #ifndef NXAGENT_UPGRADE ... #endif
   o remove #ifdef NXAGENT_UPGRADE ... #endif clauses around code blocks and leave the
     code blocks intact
   o Move all files from nx-X11/programs/Xserver/hw/nxagent/X/ to
     nx-X11/programs/Xserver/hw/nxagent/ (don't review these files in detail, we just want the
     files copied as is).
   o Remove nx-X11/programs/Xserver/hw/nxagent/X/ and files there in

### CODE REVIEW

File: nx-X11/programs/Xserver/hw/nxagent/Composite.c [hunk 1]
```diff
@@ -24,18 +24,8 @@
 #include "Options.h"
 #include "Windows.h"
 
-#undef  NXAGENT_UPGRADE
-
-#ifndef NXAGENT_UPGRADE
-
-#include "NXcomposite.h"
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/Composite.c:31]: Uhm... wait. This is undefining `NXAGENT_UPGRADE` and then pulling in `NXcomposite.h` if it's undefined.

So the old code does basically only ever `#include "NXcomposite.h"`.

`nx-X11/programs/Xserver/hw/nxagent/NXcomposite.h` is somewhat functionally equivalent to `nx-X11/lib/Xcomposite/Xcomposite.h`, although the NX version does include one more "NX" file and defines `XserverRegion` prior to that.

Nothing seems currently to include `Xcomposite.h` and `composite.h` is probably not enough.

As such, you should probably keep the `NXcomposite.h` include here.

File: nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c [hunk 2]
```diff
@@ -1,359 +0,0 @@
-#ifndef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c:1]: Removing that file is a bad idea! It's not equivalent to `composite.c`! It's somewhat equivalent to `Xcomposite.c`, but has modifications by NoMachine.

Please do not delete this file.

You may delete the `#ifndef NXAGENT_UPGRADE` guard, though, probably.

File: nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c [hunk 3]
```diff
@@ -1,359 +0,0 @@
...
-
-    #ifdef NXAGENT_SERVER
-
-    return region;
-
-    #else
-
-    return region;
-
-    #endif
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c]: You can also replace this whole thing with just `return region;`.

File: nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c (hunk 2 above)
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c:1]: Actually, disregard my previous comment. Deleting this file is probably okay. It's not referenced in `nx-X11/programs/Xserver/hw/nxagent/Imakefile` and non-empty if `NXAGENT_UPGRADE` is not set, which is not the case with recent NX.

File: nx-X11/programs/Xserver/hw/nxagent/NXcomposite.h [hunk 4]
```diff
@@ -1,86 +0,0 @@
-#ifndef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXcomposite.h:1]: Deleting **this** file is a bad idea, though.

File: nx-X11/programs/Xserver/hw/nxagent/NXcompositeext.h [hunk 5]
```diff
@@ -1,55 +0,0 @@
-#ifndef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXcompositeext.h:1]: Also a bad idea to delete this one.

File: nx-X11/programs/Xserver/hw/nxagent/NXcompositeint.h [hunk 6]
```diff
@@ -1,76 +0,0 @@
-#ifndef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXcompositeint.h:1]: Deleting this seems to be fine.

File: nx-X11/programs/Xserver/hw/nxagent/NXcompositeproto.h [hunk 7]
```diff
@@ -1,148 +0,0 @@
-#ifndef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXcompositeproto.h:1]: Deleting this seems to be fine.

File: nx-X11/programs/Xserver/hw/nxagent/NXdamage.c [hunk 8]
```diff
@@ -1,5 +1,2073 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXdamage.c]: OK, merging `nx-X11/programs/Xserver/hw/nxagent/NXdamage.c` and `nx-X11/programs/Xserver/hw/nxagent/X/NXdamage.c` sounds sane.

File: nx-X11/programs/Xserver/hw/nxagent/NXdispatch.c [hunk 9]
```diff
@@ -1,9 +1,3 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXdispatch.c]: Ok, merging `nx-X11/programs/Xserver/hw/nxagent/NXdispatch.c` and `nx-X11/programs/Xserver/hw/nxagent/X/NXdispatch.c` sounds sane.

File: nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c [hunk 10]
```diff
@@ -1,9 +1,3 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c]: Ok: merging `nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c` with `nx-X11/programs/Xserver/hw/nxagent/X/NXdixfonts.c`.

File: nx-X11/programs/Xserver/hw/nxagent/NXevents.c [hunk 11]
```diff
@@ -1,9 +1,3 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXevents.c]: OK: merging `nx-X11/programs/Xserver/hw/nxagent/NXevents.c` with `nx-X11/programs/Xserver/hw/nxagent/X/NXevents.c`

File: nx-X11/programs/Xserver/hw/nxagent/NXextension.c [hunk 12]
```diff
@@ -1,9 +1,3 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXextension.c]: OK: merging `nx-X11/programs/Xserver/hw/nxagent/NXextension.c` with `nx-X11/programs/Xserver/hw/nxagent/X/NXextension.c`.

File: nx-X11/programs/Xserver/hw/nxagent/NXglxext.c [hunk 13]
```diff
@@ -1,10 +1,21 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXglxext.c]: OK: merging `nx-X11/programs/Xserver/hw/nxagent/NXglxext.c` with `nx-X11/programs/Xserver/hw/nxagent/X/NXglxext.c`.

File: nx-X11/programs/Xserver/hw/nxagent/NXglyph.c [hunk 14]
```diff
@@ -1,9 +1,3 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXglyph.c]: OK: merging `nx-X11/programs/Xserver/hw/nxagent/NXglyph.c` with `nx-X11/programs/Xserver/hw/nxagent/X/NXglyph.c`.

File: nx-X11/programs/Xserver/hw/nxagent/NXglyphcurs.c [hunk 15]
```diff
@@ -1,9 +1,3 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXglyphcurs.c]: OK: merging `nx-X11/programs/Xserver/hw/nxagent/NXglyphcurs.c` with `nx-X11/programs/Xserver/hw/nxagent/X/NXglyphcurs.c`.

File: nx-X11/programs/Xserver/hw/nxagent/NXglyphstr.h [hunk 16]
```diff
@@ -1,9 +1,3 @@
-#ifdef NXAGENT_UPGRADE
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXglyphstr.h]: OK: merging `nx-X11/programs/Xserver/hw/nxagent/NXglyphstr.h` with `nx-X11/programs/Xserver/hw/nxagent/X/NXglyphstr.h`.

File: nx-X11/programs/Xserver/hw/nxagent/NXxvdisp.c [hunk 17]
```diff
@@ -2274,4 +2292,4 @@ void XineramifyXv(void)
 
 #endif
 
-#endif /* #ifdef NXAGENT_UPGRADE */
+#endif
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/NXxvdisp.c]: No newline at EOF and would change to

``` C
#endif /* !defined(__sun) && !defined(__CYGWIN__) */
```

**Comment by Ionic** [comment]
I've cleaned up my comments a little bit. Sorry for spamming with temporary stuff while I went through the diff.

File: nx-X11/programs/Xserver/hw/nxagent/Composite.c (hunk 1 above)
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Composite.c:31]: > So the old code does basically only ever #include "NXcomposite.h".

It does include NXcomposite.h always. That is correct, indeed. But if you look at NXcomposite.h, everything in it is embraced by 

```
#ifndef NXAGENT_UPGRADE
[...]
#endif /* #ifndef NXAGENT_UPGRADE */
```

So, we can also not include it right away and remove NXcomposite.h.

File: nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c (hunk 2 above)
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/NXcomposite.c:1]: The code in NXcomposite.c is also embraced by this:

```
#ifndef NXAGENT_UPGRADE
[...]
#endif /* #ifndef NXAGENT_UPGRADE */
```

So, removing it is sane here, IMHO.

File: nx-X11/programs/Xserver/hw/nxagent/NXcomposite.h (hunk 4 above)
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/NXcomposite.h:1]: Why? We previously built nxagent with NXAGENT_UPGRADE being defined, so we can drop all passages that would be built with NXAGENT_UPGRADE being undefined, which is exactly the case for the above quoted code line.

File: nx-X11/programs/Xserver/hw/nxagent/NXcompositeext.h (hunk 5 above)
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/NXcompositeext.h:1]: Dito.

File: nx-X11/programs/Xserver/hw/nxagent/NXxvdisp.c (hunk 17 above)
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/NXxvdisp.c]: Fixed with the next force-push on this branch.

**Comment by sunweaver** [comment]
@Ionic: Thaniks for taking your time and review this PR.

Could you please take a second look at the NXcomposite.c, NXcomposite.h, etc. files? I am not sure your reviewing has been correct there.

Note that we keep code for NXAGENT_UPGRADE being defined and drop code for NXAGENT_UPGRADE being not defined.

**Comment by Ionic** [comment]
Cannot merge this PR cleanly, because you have pushed another commit to the 3.6.x branch and your PR is not based on that.

Please rebase against current 3.6.x.

**Pull request merged**